src/main/java/core/basesyntax/io/impl/CsvReportSerializer.java

```
@@ -0,0 +1,39 @@
+package core.basesyntax.io.impl;
+
+import core.basesyntax.io.ReportSerializer;
+import core.basesyntax.model.FruitType;
+import core.basesyntax.model.Report;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CsvReportSerializer implements ReportSerializer {
+    @Override
+    public void save(Report report, Path path) {
+        List<String> csvRows = createCsvRows(report);
+        try (BufferedWriter writer = Files.newBufferedWriter(path)) {
+            for (String line : csvRows) {
+                writer.write(line);
+                writer.newLine();
+            }
+            writer.flush();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Violation: the catch block rethrows the caught IOException as `new RuntimeException(e)` without an informative message. The checklist requires not to ignore exceptions and to rethrow with an informative message and the exception object (see: "Don't ignore exceptions in try-catch blocks"). Provide a message that includes the path being written and wrap the original exception, for example: `throw new RuntimeException("Can't write report to path: " + path, e);`. This class is `CsvReportSerializer` so update the catch accordingly.